packages/cypher-builder/src/procedures/CypherProcedure.ts

```
@@ -0,0 +1,84 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Clause } from "../clauses/Clause";
+import { CypherASTNode } from "../CypherASTNode";
+import type { CypherEnvironment } from "../Environment";
+import type { Expr } from "../types";
+import { compileCypherIfExists } from "../utils/compile-cypher-if-exists";
+import type { YieldProjectionColumn } from "./Yield";
+import { Yield } from "./Yield";
+
+/** Represents a Cypher Procedure that does not yield columns
+ * @see [Cypher Documentation](https://neo4j.com/docs/cypher-manual/current/clauses/call/)
+ * @group Expressions
+ * @category Procedures
+ */
+export class VoidCypherProcedure extends Clause {
+    protected name: string;
+    private params: Array<Expr>;
+
+    constructor(name: string, params: Array<Expr> = []) {
+        super();
+        this.name = name;
+        this.params = params;
+        for (const param of params) {
+            if (param instanceof CypherASTNode) {
+                this.addChildren(param);
+            }
+        }
+    }
+
+    /** @internal */
+    public getCypher(env: CypherEnvironment): string {
+        const procedureCypher = this.getProcedureCypher(env);
+        return `CALL ${procedureCypher}`;
+    }
+
+    private getProcedureCypher(env: CypherEnvironment): string {
+        const argsStr = this.params.map((expr) => expr.getCypher(env)).join(", ");
+
+        return `${this.name}(${argsStr})`;
+    }
+}
+
+/** Represents a Cypher Procedure
+ * @see [Cypher Documentation](https://neo4j.com/docs/cypher-manual/current/clauses/call/)
+ * @group Expressions
+ * @category Procedures
+ */
+export class CypherProcedure<T extends string = string> extends VoidCypherProcedure {
+    private yieldStatement: Yield | undefined;
+
+    public yield(...columns: Array<"*" | YieldProjectionColumn<T>>): Yield {
+        if (columns.length === 0) throw new Error("Empty projection in CALL ... YIELD");
```

Do we want to raise an error if the `columns.length` is zero or treat this as *? Potentially having a default wildcard for an empty list could improve the ergonomics of the users that pass directly the columns from user input to the yield method without asking them to verify if the length is zero.
For instance
```typescript
const projectedColumn = someUserInput;
new CypherProcedure("db.index.fulltext.listAvailableAnalyzers").yield(projectedColumns)
```
rather than:
```typescript
const projectedColumn = someUserInput.length ? someUserInput : "*";
new CypherProcedure("db.index.fulltext.listAvailableAnalyzers").yield(projectedColumns)
```

---

I'd rather make it explicit tbh, this way we can stay close to the generated cypher and the user will not get unexpected results.